atlp-rwanda/phantom-be-dream-team
# # 10feature: Register Drivers and Operators

What does this PR do?

register drivers and operators

Description of Task to be completed?


1)The driver or operator should be able to sign in to the app using their email and the provided password.
2)Appropriate error messages are displayed on unsuccessful attempts
3)The administrator should be able to register a new driver or a new operator.
4)An email containing a random password should be sent to the provided email.


Have the following endpoint working

POST /api/v1/users/register

How should this be manually tested?

git pull this branch
run npm install
check env.example file and add DEV_DATABASE_URL and JWT_SECRET  and USER_EMAIL IN to your .env file
Do migrations with npm run migrate
run npx sequelize-cli db:seed:all to create pre-configured credentials for seeders
run npm run start
Test the route POST /users/register in POSTMAN or any other platform for using API
Any background context you want to provide?
I used the JWT package for hashing the password stored in the Database

What are the relevant pivotal tracker stories?

https://trello.com/c/jcZ6Wev3/10-register-drivers-operators
![WhatsApp Image 2022-06-29 at 2 02 57 PM](https://user-images.githubusercontent.com/88536742/176440382-74ece79c-3047-4dbe-b3ba-e5170b2c2f54.jpeg)

Questions:
No.
